homeassistant/components/random/config_flow.py

```diff
@@ -0,0 +1,225 @@
+"""Config flow for Random helper."""
+from collections.abc import Callable, Coroutine, Mapping
+from typing import Any, cast
+
+import voluptuous as vol
+
+from homeassistant.components.binary_sensor import BinarySensorDeviceClass
+from homeassistant.components.sensor import DEVICE_CLASS_UNITS, SensorDeviceClass
+from homeassistant.const import (
+    CONF_DEVICE_CLASS,
+    CONF_MAXIMUM,
+    CONF_MINIMUM,
+    CONF_NAME,
+    CONF_UNIT_OF_MEASUREMENT,
+    Platform,
+)
+from homeassistant.core import callback
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.schema_config_entry_flow import (
+    SchemaCommonFlowHandler,
+    SchemaConfigFlowHandler,
+    SchemaFlowFormStep,
+    SchemaFlowMenuStep,
+)
+from homeassistant.helpers.selector import (
+    SelectSelector,
+    SelectSelectorConfig,
+    SelectSelectorMode,
+    TextSelector,
+)
+
+from .const import DOMAIN
+from .sensor import DEFAULT_MAX, DEFAULT_MIN
+
+NONE_SENTINEL = "none"
+
+
+def generate_schema(domain: str, flow_type: str) -> dict[vol.Marker, Any]:
+    """Generate schema."""
+    schema: dict[vol.Marker, Any] = {}
+
+    if domain == Platform.BINARY_SENSOR and flow_type == "config":
+        schema = {
+            vol.Optional(CONF_DEVICE_CLASS): SelectSelector(
+                SelectSelectorConfig(
+                    options=[
+                        NONE_SENTINEL,
+                        *sorted(
+                            [cls.value for cls in BinarySensorDeviceClass],
+                            key=str.casefold,
+                        ),
+                    ],
```

Comment:
```suggestion
                    options=[cls.value for cls in BinarySensorDeviceClass],
                    sort=True,
```

We now have an optional selectSelector. See also https://github.com/home-assistant/core/pull/101279